### PR TITLE
Complete rewrite with docker-py and click

### PR DESCRIPTION
**Pull request by heucke**
* Drop isolate ordering options with the build subcommand

Signed-off-by: Tyler Heucke <tyler@heucke.io>

### CODE REVIEW

**Comment by heucke** [comment]
Pull-request updated, HEAD is now 74ec1d207e0c0ed9e9ff1ed490d5baa4d22d8531

File: dbp.py [hunk 1]
```diff
@@ -2,638 +2,338 @@
...
-    return template.format(image, IMAGE_VERSION, dist)
-
-
-def docker_pull_images(image: str, dist: str, check_first=False) -> int:
-    """Runs docker pull for both build and development images and returns the return code"""
-    if ":" in image:
-        # manually specified image tag, assume no pull needed
+    if not targets and ws.path == Path.cwd():
+        # If run from workspace root with no targets, build all in dependency order
+        targets = tuple(n for n in nx.dfs_postorder_nodes(ws.builddepends_graph()))
```
Comment (42B) [inline dbp.py]:
```python
targets = tuple(nx.dfs_postorder_nodes(ws.builddepends_graph()))  # ?
```

**Comment by heucke** [comment]
Pull-request updated, HEAD is now ff5ed9aa7aa258797cc47fca5a9331425fa5b5db

**Comment by heucke** [comment]
Pull-request updated, HEAD is now 5587c0f6ad49430455addfaee4702d09fa6ffa41

File: dbp.py (hunk 1 above)
Comment (heucke) [reply dbp.py]: Thanks :)

**Comment by heucke** [comment]
Pull-request updated, HEAD is now 5ff84aab57710d0c674c492529032f714743b456

**Comment by heucke** [comment]
Pull-request updated, HEAD is now fe296e0f454a79a592d301be51db3980dd988b5d

**Comment by heucke** [comment]
Pull-request updated, HEAD is now ca7941aaaed37257dd57806a928af1ae790df4a6

**Comment by heucke** [comment]
Pull-request updated, HEAD is now 3ab3914aafa5b08a6c5bf1d1c45adf05965eaae8

**Comment by heucke** [comment]
Pull-request updated, HEAD is now d084bc74fb0dd213683efd7a4b622eae6afbbd0d

**Comment by heucke** [comment]
Pull-request updated, HEAD is now 159db20b14acf2c3a0ecb53aad5ec90d6fe5ca15

**Pull request merged**